### PR TITLE
Deleting ProvisioningStateSpecifiedForLRODelete rule as it is incorrect

### PR DESCRIPTION
**Pull request by rkmanda**
[Task 22895741](https://msazure.visualstudio.com/One/_workitems/edit/22895741): Disable ProvisioningStateSpecifiedForLRODelete

[ProvisioningStateSpecifiedForLRODelete](https://github.com/Azure/azure-openapi-validator/blob/main/docs/provisioning-state-specified-for-lro-delete.md)

A long-running delete doesnt have a response body so this rule is checking for the existence of the provisioningState property in the 200 response body incorrectly. The original implementation was due to a misunderstanding that a 200 status code should be specified for a long-running delete operation which in itself s incorrect.

### CODE REVIEW

File: common/changes/@microsoft.azure/openapi-validator-rulesets/rkmanda-ProvisioningStateSpecifiedForLRODelete_2023-06-01-05-44.json
Comment (bdefoy) [inline common/changes/@microsoft.azure/openapi-validator-rulesets/rkmanda-ProvisioningStateSpecifiedForLRODelete_2023-06-01-05-44.json:1]: Delete this file. You have two changelog files here and you removed the code, did not disable the rule

**Review by bdefoy — APPROVED** [review]

**Comment by tejaswiMinnu** [comment]
I see some warnings of "ProvisioningStateSpecifiedForLRODelete" rule in CONTRIBUTING.md 
Do we have to remove those as well?

**Review by AkhilaIlla — APPROVED** [review]

**Review by AkhilaIlla — APPROVED** [review]

File: common/changes/@microsoft.azure/openapi-validator-rulesets/rkmanda-ProvisioningStateSpecifiedForLRODelete_2023-06-01-05-44.json
Comment (rkmanda) [reply common/changes/@microsoft.azure/openapi-validator-rulesets/rkmanda-ProvisioningStateSpecifiedForLRODelete_2023-06-01-05-44.json:1]: done

**Comment by rkmanda** [comment]
> I see some warnings of "ProvisioningStateSpecifiedForLRODelete" rule in CONTRIBUTING.md Do we have to remove those as well?

Thats just an example of what the output looks like so its ok to leave it as such.

**Pull request merged**